src/ImageProcessor.Web/Caching/DiskCache.cs

```diff
@@ -457,6 +460,49 @@ private static string GetValidatedAbsolutePath(string originalPath, out string v
         }
 
         /// <summary>
+        /// Returns a value indicating whether the requested image has been updated.
+        /// </summary>
+        /// <param name="creationDate">The creation date.</param>
+        /// <returns>The <see cref="bool"/></returns>
+        private async Task<bool> IsUpdatedAsync(DateTime creationDate)
+        {
+            bool isUpdated = false;
+
+            try
+            {
+                if (new Uri(this.RequestPath).IsFile)
+                {
+                    FileInfo imageFileInfo = new FileInfo(this.RequestPath);
```

Comment:
Instead of allocating a FileInfo instance which it seems you don't need you can achieve the same thing using the static System.IO.File methods to get the Exists and Utc date info from the file